File: Estruturas e bibliotecas/1709/1709.py
metadeTamanhoBaralho = int(input())/2
contador, posicaoZero = 1, 1

while True:
    contador += 1
    
    if(posicaoZero < metadeTamanhoBaralho):
        pos = posicaoZero
        posicaoZero = pos*2+1
    else:
        pos = posicaoZero - metadeTamanhoBaralho
        if(pos == 0):
            break
        posicaoZero = pos * 2
        
print(contador)

File: Estruturas e bibliotecas/1709/1709.cpp
#include <iostream>

using namespace std;

int main()
{
    int metadeTamanhoBaralho, contador = 1, posicaoZero = 1, pos;
    cin >> metadeTamanhoBaralho;
    metadeTamanhoBaralho /= 2;
    
    while(1) {
        contador++;
        
        if(posicaoZero < metadeTamanhoBaralho) {
            pos = posicaoZero;
            posicaoZero = pos*2+1;
        }
        else {
            pos = posicaoZero - metadeTamanhoBaralho;
            if (pos == 0) {
                break;
            }
            posicaoZero = pos * 2;
        }
    }
    cout << contador << "\n";

    return 0;
}
